cli/default.php

```diff
@@ -0,0 +1,105 @@
+
+/**
+ * CLI script for default settings value for activity.
+ *
+ * @package    local_integrity
+ * @copyright  2025 Catalyst IT
+ * @author     Guillaume BARAT (guillaumebarat@catalyst-au.net)
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+use local_integrity\plugininfo\integritystmt;
+use local_integrity\statement_factory;
+
+define('CLI_SCRIPT', true);
+
+require(__DIR__ . '/../../../config.php');
+require_once($CFG->libdir . '/clilib.php');
+
+[$options, $unrecognized] = cli_get_params(
+    [
+        'default' => false,
+        'help' => false,
+    ],
+    [
+        'd' => 'default',
+        'h' => 'help',
+    ]
+);
+
+if ($unrecognized) {
+    $unrecognized = implode("\n  ", $unrecognized);
+    cli_error(get_string('cliunknowoption', 'admin', $unrecognized));
+}
+
+if ($options['help']) {
+    $help = <<<EOT
+Default value for academic integrity.
+
+Options:
+ -d, --default             Set default integrity settings for each activity that has not been set yet.
+ -h, --help                Print out this help
+
+Example:
+\$sudo -u www-data /usr/bin/php local/integrity/cli/default.php --default=0,1
+
+EOT;
+    cli_writeln($help);
+    exit(0);
+}
+
+if (isset($options['default']) && in_array($options['default'], [0, 1])) {
+    $integritytable = \local_integrity\settings::TABLE;
+    $enabled = $options['default'];
+    $stmt = statement_factory::get_statements();
+    $pluginlist = integritystmt::get_enabled_plugins();
+    $pluginstmt = [];
+    foreach ($pluginlist as $name) {
+        $pluginstmt[$name] = $stmt[$name]->get_plugin_name();
+    }
+    [$insqlplugin, $paramsplugins] = $DB->get_in_or_equal(array_keys($pluginlist));
+    $sql = "SELECT ct.id contextid, m.name modulename
+                          FROM {modules} m
+                    INNER JOIN {course_modules} cm ON m.id = cm.module
+                    INNER JOIN {context} ct ON ct.instanceid = cm.id
+                     LEFT JOIN {" . $integritytable . "} lis ON lis.contextid = ct.id
+                         WHERE ct.contextlevel = 70
+                           AND m.name $insqlplugin
+                           AND lis.contextid IS NULL";
+    $datacontextplugins = $DB->get_recordset_sql($sql, $paramsplugins);
+    $buildobject = [];
+    foreach ($datacontextplugins as $datacontext) {
+        $buildobject[] = (object) [
+                'id' => null,
+                'contextid' => $datacontext->contextid,
+                'plugin' => $pluginstmt[$datacontext->modulename],
+                'enabled' => $enabled,
+                'usermodified' => 2,
+                'timecreated' => time(),
+                'timemodified' => time(),
+        ];
+    }
+    $datacontextplugins->close();
+    // Split the object in batch of 2000.
+    $chunks = array_chunk($buildobject, 2000);
+    foreach ($chunks as $chunk) {
+        $DB->insert_records($integritytable, $chunk);
+    }
+    cli_writeln("Activity default setting all set.");
```

Comment:
Not critical, but would be good to let the user know how  many records was updated? Like

` cli_writeln("Default setting of 1 was set for 1000 activities.");`

or something like that